data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/PstRangeActionCreator.java

```diff
@@ -0,0 +1,95 @@
+        CsaProfileCracUtils.checkNormalEnabled(tapPositionActionPropertyBag, remedialActionId, "TapPositionAction");
+        CsaProfileCracUtils.checkPropertyReference(tapPositionActionPropertyBag, remedialActionId, "TapPositionAction", CsaProfileConstants.PROPERTY_REFERENCE_TAP_POSITION);
+        String rawId = tapPositionActionPropertyBag.get(CsaProfileConstants.TAP_CHANGER);
+        String tapChangerId = rawId.substring(rawId.lastIndexOf("_") + 1);
+        IidmPstHelper iidmPstHelper = new IidmPstHelper(tapChangerId, network);
+        if (!iidmPstHelper.isValid()) {
+            throw new FaraoImportException(ImportStatus.ELEMENT_NOT_FOUND_IN_NETWORK, "Remedial Action: " + remedialActionId + " will not be imported because " + iidmPstHelper.getInvalidReason());
+        }
+
+        pstRangeActionAdder
+                .withNetworkElement(tapChangerId)
+                .withInitialTap(iidmPstHelper.getInitialTap())
+                .withTapToAngleConversionMap(iidmPstHelper.getTapToAngleConversionMap());
+
+        if (!linkedStaticPropertyRangesFoTapPositionAction.isEmpty()) {
+            Optional<Integer> normalValueUp = Optional.empty();
+            Optional<Integer> normalValueDown = Optional.empty();
+            for (PropertyBag staticPropertyRangePropertyBag : linkedStaticPropertyRangesFoTapPositionAction) {
+                CsaProfileCracUtils.checkPropertyReference(staticPropertyRangePropertyBag, remedialActionId, "StaticPropertyRange", CsaProfileConstants.PROPERTY_REFERENCE_TAP_POSITION);
+                String valueKind = staticPropertyRangePropertyBag.get(CsaProfileConstants.STATIC_PROPERTY_RANGE_VALUE_KIND);
+
+                if (!valueKind.equals(CsaProfileConstants.VALUE_KIND_ABSOLUTE)) {
+                    throw new FaraoImportException(ImportStatus.ELEMENT_NOT_FOUND_IN_NETWORK, "Remedial Action: " + remedialActionId + " will not be imported because StaticPropertyRange has wrong value of valueKind, the only allowed value is 'absolute'");
+                } else {
+                    String direction = staticPropertyRangePropertyBag.get(CsaProfileConstants.STATIC_PROPERTY_RANGE_DIRECTION);
+                    int normalValue = (int) Float.parseFloat(staticPropertyRangePropertyBag.get(CsaProfileConstants.NORMAL_VALUE));
+                    if (direction.equals(CsaProfileConstants.DIRECTION_DOWN)) {
+                        normalValueDown = Optional.of(normalValue);
+                    } else if (direction.equals(CsaProfileConstants.DIRECTION_UP)) {
+                        normalValueUp = Optional.of(normalValue);
+                    } else {
+                        throw new FaraoImportException(ImportStatus.ELEMENT_NOT_FOUND_IN_NETWORK, "Remedial Action: " + remedialActionId + " will not be imported because StaticPropertyRange has wrong value of direction, the only allowed values are RelativeDirectionKind.up and RelativeDirectionKind.down");
```

Comment:
```suggestion
                        throw new FaraoImportException(ImportStatus.NOT_YET_HANDLED_BY_FARAO, "Remedial Action: " + remedialActionId + " will not be imported because StaticPropertyRange has wrong value of direction, the only allowed values are RelativeDirectionKind.up and RelativeDirectionKind.down");
```